gplugins/design_recipe/DesignRecipe.py

```diff
@@ -25,57 +27,64 @@ class DesignRecipe:
     # the hash of the system last time eval() was executed
     last_hash: int
 
-    # The component this DesignRecipe operates on. This is not necessarily
-    # the same `component` refered to in the `dependencies` recipes.
-    component: Optional[Component] = None
+    # The component factory this DesignRecipe operates on. This is not necessarily
+    # the same `component` referred to in the `dependencies` recipes.
+    cell: ComponentFactory | Component | None = None
 
     # LayerStack for the process that the component is generated for
     layer_stack: LayerStack
 
-    def __init__(self,
-                 component: Component,
-                 layer_stack: LayerStack = get_layer_stack(),
-                 dependencies: List[dr.DesignRecipe] = []):
+    def __init__(
+        self,
+        cell: ComponentFactory | Component,
+        dependencies: list[dr.DesignRecipe] | None = None,
+        layer_stack: LayerStack = get_layer_stack(),
+    ):
+        dependencies = dependencies or []
         self.dependencies = dr.ConstituentRecipes(dependencies)
-        self.component = component
+        self.cell = cell
         self.last_hash = -1
         self.layer_stack = layer_stack
 
     def __hash__(self) -> int:
         """
         Returns a hash of all state this DesignRecipe contains.
-        Subclasses should include functionality-specific state (e.g. fdtd settings) here. 
+        Subclasses should include functionality-specific state (e.g. fdtd settings) here.
         This is used to determine 'freshness' of a recipe (i.e. if it needs to be rerun)
+
+        Hashed items:
+        - component or cell
+        - layer stack
         """
         h = hashlib.sha1()
-        if (self.component is not None):
-            h.update(
-                self.component.hash_geometry(precision=1e-4).encode('utf-8'))
-        h.update(
-            self.layer_stack.hash_geometry(precision=1e-4).encode('utf-8'))
-        return int.from_bytes(h.digest(), 'big')
+        if self.cell is not None:
+            if isinstance(self.cell, Callable):
+                h.update(self.cell().hash_geometry(precision=1e-4).encode("utf-8"))
+            elif type(self.cell) == Component:
+                h.update(self.cell.hash_geometry(precision=1e-4).encode("utf-8"))
+        h.update(self.layer_stack.model_dump_json().encode("utf-8"))
+        return int.from_bytes(h.digest(), "big")
 
     def is_fresh(self) -> bool:
         """
         Returns if this DesignRecipe needs to be re-`eval()`ed.
         This could be either caused by this DesignRecipe's
         configuration being changed, or that of one of its dependencies.
         """
-        return hash(self) == self.last_hash and \
-            all([recipe.is_fresh() for recipe in self.dependencies])
+        return hash(self) == self.last_hash and all(
+            recipe.is_fresh() for recipe in self.dependencies
+        )
 
     def eval(self, force_rerun_all=False) -> bool:
```

Comment:
oops there should be a comment on `force_rerun_all` (it forces this DR and its dependencies to re-`eval()` regardless of freshness)